src/gc/gc.cpp

```diff
@@ -15920,7 +15952,8 @@ start_no_gc_region_status gc_heap::prepare_for_no_gc_region (uint64_t total_size
 
     if (allocation_no_gc_soh != 0)
     {
-        current_no_gc_region_info.soh_allocation_size = allocation_no_gc_soh;
+        assert(allocation_no_gc_soh <= SIZE_T_MAX);
```

Comment:
>assert(allocation_no_gc_soh <= SIZE_T_MAX); [](start = 8, length = 43)

you already asserted total_allowed is <= SIZE_T_MAX so I don't see why they are checked again here.

Reply:
just extra-defensive coding - no reason in particular. I do see that I typo'd one of the asserts, though, on line 15972, so if they're not useful I'll just delete them.

Reply:
looks like I'm also missing a `allocation_no_gc_loh > total_allowed_soh_allocation` check, so I'll add that in the next commit.

Reply:
yeah I would remove them.